### ISSUE TITLE
Improvement: Option to disable element "Updating" with Element creation nodes

### ISSUE THREAD

**Issue by AndrewHeumann** [DynamoDS/DynamoRevit#1673]
many folks including myself have complained about this issue: https://forum.dynamobim.com/t/circumvent-revit-object-replacement/7459/14
https://forum.dynamobim.com/t/dynamo-is-remembering-the-previous-generated-or-changed-elements-and-deleting-them/7491/2

The current default element binding behavior is generally useful and should be preserved. However, there are a number of use cases in which it is undesirable. For instance:

* a user wants to create a re-usable tool to repeatedly create views based on text inputs, as described in the first link, 
* a user wants to create building pads from floor outlines where the user will manually and sequentially select a floor at a time in the revit model
* user wants to select an instance of a table family and then specify a chair count and have chair families automatically placed - then select another table and repeat w/ different chair count, etc.

A right-click option - on a per-node basis, or perhaps at a graph level (I think per-node is better) - should enable the user to disable the aspect of binding behavior that deletes/updates created elements, instead just creating a new instance of the element in question every time the graph runs. I can also imagine wanting to enforce "manual run mode" in a case like this such that someone, say, dragging a slider wouldn't accidentally generate many copies of the same element. 

The current workarounds - requiring scripting, or quitting and restarting as described in the threads linked earlier are very far from workable. Adding this option would open the door to all kinds of useful "creation utilities" as I've described above. I can also imagine a really nice integration with dynamo player down the road.

**Comment by jnealb** [comment]
@kronz @riteshchandawar Ping

**Comment by ksobon** [comment]
@AndrewHeumann this has been the "thing" since Element Binding was introduced. I think you can "disable" it, if you manage to never use Dynamo elements. Think of it this way. If you don't pass out the elements and don't wrap them into `DSType`, then you can create family instances that are not bound and will behave like the regular ones. The minute you start wrapping them, they will get bound. I think that's what I was able to do once before. It's not great.

**Comment by AndrewHeumann** [comment]
@ksobon Yeah. If you're not scripting you're basically SOL.

**Comment by andydandy74** [comment]
Shouldn't be an issue with Player though, right?

**Comment by kronz** [comment]
@andydandy74 Yes, with Dynamo Player there is effectively no element binding because there is no multiple run during a session and no saving of the state of the graph after run.  So, this is effective for some workflows, but certainly not all. 

In terms of how this could be implemented, a few more considerations:
- Advantages and disadvantages of an Application level setting, Document setting, or Node setting to allow or disallow binding  
- Are there times when you want to have element binding and then at some point disconnect created elements from the nodes that created them (argues for a per-node "flush" or "clear" mechanism)
- Visibility into what the state of any particular node is, in terms of it using or not using Element binding
- Relationship to Auto v. Manual.  As Andrew points out . . . seems like you really don't want Auto and no element binding, but are there cases you would?  I could see some applications with Periodic update
- Conceptual and Interaction differences between "baking" out Revit elements from nodes and clearing element binding.  
- Would there be a case for "re-attaching" binding to an element that was created?

**Comment by sixtysecondrevit** [comment]
Might be a little off-topic and not a perfect solution. But, I added a "bake" node in Beaker to achieve this result when I am finished creating elements. Essentially it copies the element in place and deletes the "attached to Dynamo" one through the use of a toggle. Perhaps the Dynamo team provided solution can be another node that breaks the relationship?
![20170523-bakeelements](https://cloud.githubusercontent.com/assets/15744724/26377402/525b8f2c-3fcd-11e7-86ef-b3a06265a035.gif)

**Comment by andydandy74** [comment]
Like an Element.RemoveBinding node? That would be sweet!

**Comment by jnealb** [comment]
Pandora or Icarus? Which myth is best here?

**Comment by sixtysecondrevit** [comment]
@kronz 
_Advantages and disadvantages of an Application level setting, Document setting, or Node setting to allow or disallow binding_
**I see more disadvantages as people would forget that this feature is activated**

_Are there times when you want to have element binding and then at some point disconnect created elements from the nodes that created them (argues for a per-node "flush" or "clear" mechanism)_
**When I am done with my workflow and may not need to revisit or need to use to create new stuff.**

_Would there be a case for "re-attaching" binding to an element that was created?_
**I think I would just recreate the elements if I already made the decision to detach it.**

**Comment by archie456** [comment]
Having persistent bindings is a real issue for me, and is one of the problems of why I can't simply give out graphs to the people I work with.

Having a Element.RemoveBinding node or something similar would be perfect!

**Comment by kennyb6** [comment]
A button next to the Run that just lets you to detach and commit what you have done would be awesome. I am pretty new to Dynamo but one graph I have been using often is to place lighting fixtures around a cove using select edge, due to the cove being a generic model of varying sizes. Because select edge stops when a line changes to an arc or something else, I have to run the script multiple times to be able to do a full floor's lighting.

This involves a lot of testing of offsets to make sure everything is flush and continuous. My current workflow is once I get a section/curve lights properly positioned, I have to copy and paste to same position. It would be much smoother if there was a button I could press after running and checking the position, to just detach and commit it. I don't necessarily want a node (though a simple boolean would be similar), just some way to commit the changes after running the script and checking everything.

**Comment by August-R** [comment]
I'm just going to add my voice to the list of voices here. Coming from a Rhino/Grasshopper background, a "Commit Changes"/"Bake"/"Detach Element"/etc is sorely needed. For me, it's consistently an issue of wanting to run a script multiple times, but needing to change certain inputs each time. What I really want is a way to tinker with the script (i.e. change some variables), commit the changes and bake the geometry, and then move on to the next set of tinkering. Right now I need to close and re-open the script each time, which is just unnecessary.

I think kennyb6's suggestion, with a button to "detach" next to the Run button is the cleanest way, at least from a usability standpoint (no idea what it may entail under the hood). I appreciate all the work-arounds posted here, but when your goal is speed and efficiency (the reason I'm using Dynamo in the first place), a work-around takes time and doesn't actually solve the problem. 

I understand the reasons for not having such a feature... but at the same time, I am a bit surprised such a feature is not in the software and people here seem to think it isn't necessary. Consider this an emphatic message of support for such a feature.
